sova/supervisor/progression.py

```
@@ -1215,14 +1214,14 @@ async def _check_repeated_failures_gate(self, issue: int) -> BlockReason | None:
                     return BlockReason(
-                        gate="repeated_failure",
+                        gate=gate,
                         detail=(
-                            f"Researcher has failed {count} times for #{issue}; "
+                            f"{role.capitalize()} has failed {count} times for #{issue}; "
                             f"human review required (threshold: {max_failures})"
                         ),
                     )
         except Exception:
-            log.debug("repeated_failures.check_failed", issue=issue, exc_info=True)
+            log.debug(f"{gate}.check_failed", issue=issue, exc_info=True)
```

Ruff (0.16.1)

[warning] Logging statement uses f-string

(G004)